### ISSUE TITLE
GPlay:   java.lang.IllegalStateException in com.owncloud.android.providers.FileContentProvider.delete

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#1319]
Google Play Store:

java.lang.IllegalStateException
com.owncloud.android.providers.FileContentProvider.delete

* Version: 1.4.3+
* Android: 6+

```
java.lang.IllegalStateException: 
  at android.database.CursorWindow.nativeGetLong (Native Method)
  at android.database.CursorWindow.getLong (CursorWindow.java:511) 
  at android.database.AbstractWindowedCursor.getLong (AbstractWindowedCursor.java:75)
  at com.owncloud.android.providers.FileContentProvider.delete (FileContentProvider.java:146)
  at com.owncloud.android.providers.FileContentProvider.delete (FileContentProvider.java:152)
  at com.owncloud.android.providers.FileContentProvider.delete (FileContentProvider.java:152)
  at com.owncloud.android.providers.FileContentProvider.delete (FileContentProvider.java:152)
  at com.owncloud.android.providers.FileContentProvider.delete (FileContentProvider.java:95)
  at android.content.ContentProviderOperation.apply (ContentProviderOperation.java:311)
  at com.owncloud.android.providers.FileContentProvider.applyBatch (FileContentProvider.java:550)
  at android.content.ContentProvider$Transport.applyBatch (ContentProvider.java:315)
  at android.content.ContentProviderClient.applyBatch (ContentProviderClient.java:419)
  at com.owncloud.android.datamodel.FileDataStorageManager.saveFolder (FileDataStorageManager.java:405)
  at com.owncloud.android.operations.RefreshFolderOperation.synchronizeData (RefreshFolderOperation.java:441)
  at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder (RefreshFolderOperation.java:318)
  at com.owncloud.android.operations.RefreshFolderOperation.run (RefreshFolderOperation.java:198)
  at com.owncloud.android.lib.common.operations.RemoteOperation.execute (RemoteOperation.java:136)
  at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder (FileSyncAdapter.java:271)
  at com.owncloud.android.syncadapter.FileSyncAdapter.onPerformSync (FileSyncAdapter.java:179)
  at android.content.AbstractThreadedSyncAdapter$SyncThread.run (AbstractThreadedSyncAdapter.java:272)
```

**Comment by tobiasKaminsky** [comment]
Google is not saying what the real exception is.
> childId = children.getLong(children.getColumnIndex(ProviderTableMeta._ID));

I guess that somehow getColumnIndex returns "-1" as fallback and therefore getLong(-1) throws the error above (manually tested that it throws the exception).

I use a try/catch block to avoid the crash, but this would not solve the root cause.
Does anybody have an idea why 
> children.getColumnIndex(ProviderTableMeta._ID)

returns "-1"?

**Comment by AndyScherzinger** [comment]
```-1``` means the column doesn't exist on the actual cursor but I don't know why that is...

**Comment by tobiasKaminsky** [comment]
Somehow the database upgrade did not run successfully. 
To circumvent this, I could try to catch the error and then re-run a database integration check
- check if all tables exists
- check if all columns exists
  --> otherwise create them
- adjust database version

**Comment by tobiasKaminsky** [comment]
There is a fix in #1342, hopefully it will work

**Comment by AndyScherzinger** [comment]
Also keeps happing with 2.0.0 RC8, but I found a 1.4.3 Stacktrace in the old crash console:

```
java.lang.IllegalStateException: Couldn't read row 1572, col 0 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
	at android.database.CursorWindow.nativeGetLong(Native Method)
	at android.database.CursorWindow.getLong(CursorWindow.java:524)
	at android.database.AbstractWindowedCursor.getLong(AbstractWindowedCursor.java:75)
	at com.owncloud.android.providers.FileContentProvider.delete(FileContentProvider.java:146)
	at com.owncloud.android.providers.FileContentProvider.delete(FileContentProvider.java:152)
	at com.owncloud.android.providers.FileContentProvider.delete(FileContentProvider.java:152)
	at com.owncloud.android.providers.FileContentProvider.delete(FileContentProvider.java:152)
	at com.owncloud.android.providers.FileContentProvider.delete(FileContentProvider.java:95)
	at android.content.ContentProviderOperation.apply(ContentProviderOperation.java:316)
	at com.owncloud.android.providers.FileContentProvider.applyBatch(FileContentProvider.java:550)
	at android.content.ContentProvider$Transport.applyBatch(ContentProvider.java:328)
	at android.content.ContentProviderClient.applyBatch(ContentProviderClient.java:444)
	at com.owncloud.android.datamodel.FileDataStorageManager.saveFolder(FileDataStorageManager.java:405)
	at com.owncloud.android.operations.RefreshFolderOperation.synchronizeData(RefreshFolderOperation.java:441)
	at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder(RefreshFolderOperation.java:318)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:198)
	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:271)
	at com.owncloud.android.syncadapter.FileSyncAdapter.onPerformSync(FileSyncAdapter.java:179)
	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:272)
```
